`rust_binary` deps reference inconsistent library name

When a crate has both lib and bin with the same name, the generated rust_binary rule adds a dep like `:lib{name}`, but the library rule is named `{name}-lib`. This mismatch causes Buck2 to fail to resolve the dependency.

```
BUILD FAILED
Error running analysis for `root//project/slayerfs:slayerfs (cfg:no-san#df4a643de9ce91c5)`

Caused by:
    0: Error in configured node dependency, dependency chain follows (-> indicates depends on, ^ indicates same configuration as previous):
              root//project/slayerfs:slayerfs (cfg:no-san#df4a643de9ce91c5)
           -> root//project/slayerfs:libslayerfs (^)
       
    1: looking up unconfigured target node `root//project/slayerfs:libslayerfs`
    2: Unknown target `libslayerfs` from package `root//project/slayerfs`.
       Did you mean one of the 4 targets in root//project/slayerfs:BUCK?
       Maybe you meant one of these similar targets?
         root//project/slayerfs:slayerfs
       
       Available targets:
         root//project/slayerfs:slayerfs
         root//project/slayerfs:slayerfs-lib
         root//project/slayerfs:slayerfs-manifest
         root//project/slayerfs:slayerfs-vendor
```

#57

--- a/src/buckify/rules.rs
+++ b/src/buckify/rules.rs
@@ -142,7 +142,7 @@ pub fn buckify_root_node(node: &Node, ctx: &BuckalContext) -> Vec<Rule> {
             // Cargo allows `main.rs` to use items from `lib.rs` via the crate's own name by default.
             rust_binary
                 .deps_mut()
-                .insert(format!("{}-lib", bin_target.name));
+                .insert(format!(":{}-lib", bin_target.name));
         }
 
         buck_rules.push(Rule::RustBinary(rust_binary));

